components/Button.js

```diff
@@ -34,6 +36,8 @@ export default function Button({
 
   const button = (
     <button
+    href={href}
```

Comment:
```suggestion
```

Shouldn't we move the href prop to `Link`?
It seems like the intention was to use the `href` prop to determine whether to render a `Link` or a `button`. However, the return statement (`return restProps.href ? link : button;`) will always return a `link` (without an `href`) in this case.

Maybe we should move the `href` prop to the `Link` component. This way, when `href` is present, the `Link` component will be rendered with the `href` attribute, and when `href` is not present, a `button` will be rendered instead.

```
<Link
      href={href}
```
Also we should change the return statement to `href ? link : button;`

Reply:
i was thinking of the same thing as well